src/layout/ExpandableTable.js

```diff
@@ -0,0 +1,105 @@
+import _ from 'lodash/fp'
+import * as F from 'futil-js'
+import React from 'react'
+import { observable } from 'mobx'
+import { observer, inject } from 'mobx-react'
+
+let ExpandableTable = inject(() => {
+  let state = {
+    expanded: observable(new Map()),
```

Comment:
Are you using `Map` because the keys aren't strings?

Reply:
The main reason was because keys are dynamic for the each row that is expanded and the observer re-renders based on keys being added/removed from the expanded observable.

expanded: {
"KeyValue1": {},
"KeyValue5": {}
}